src/main/java/net/sf/jabref/shared/prefs/SharedDatabasePreferences.java

```diff
@@ -89,4 +108,25 @@ public void clear() throws BackingStoreException {
         internalPrefs.clear();
     }
 
+    private Optional<String> getOptionalValue(String key) {
+        return Optional.ofNullable(internalPrefs.get(key, null));
+    }
+
+    public static void clearAll() throws BackingStoreException {
+        Preferences.userNodeForPackage(JabRefMain.class).parent().node(PARENT_NODE).clear();
+    }
+
+    public void putAllDBMSConnectionProperties(DBMSConnectionProperties properties) {
```

Comment:
Why don't check here if the properties are valid? (Senn this method somewhere above...)